### PR TITLE
fix all golint and gosimple issues in kube-state-metrics

### PR DESCRIPTION
**Pull request by tariq1890**
Fix all golint issues in kube-state-metrics.

This should bring the goreportcard to 100%

### CODE REVIEW

File: internal/collector/configmap.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ var (
 		},
 		{
 			Name: "kube_configmap_created",
-			Type: metric.MetricTypeGauge,
+			Type: metric.Gauge,
```
Comment (tariq1890) [inline internal/collector/configmap.go]: This actually fixes a golint issue. I guess this is concise and self-explanatory which is typical of idiomatic Go. 

`metric.MetricTypeGauge` is long and redundant.

File: internal/collector/builder.go [hunk 2]
```diff
@@ -64,10 +64,8 @@ func NewBuilder(
 
 // WithEnabledCollectors sets the enabledCollectors property of a Builder.
 func (b *Builder) WithEnabledCollectors(c []string) {
-	copy := []string{}
-	for _, s := range c {
-		copy = append(copy, s)
-	}
+	var copy []string
+	copy = append(copy, c...)
```
Comment (tariq1890) [inline internal/collector/builder.go]: This fixes an issue reported by `gosimple`

**Comment by andyxning** [comment]
@tariq1890 Thanks for taking care of this. Could you also please add golint or gosimple checks just like `make gofmtcheck`.

**Comment by tariq1890** [comment]
@andyxning I did have a plan to introduce golangci-lint to the CI. It is a fast and efficient linter that can run multiple linters like `golint`, `gosimple` etc. I can do that in a follow-up PR.

If you think it's more appropriate to add golangci-lint to this PR, I can do that as well.

**Comment by brancz** [comment]
Let's add the CI steps in a follow up. Thanks for doing this!

/lgtm

**Pull request merged**